src/Swashbuckle.AspNetCore.SwaggerGen/XmlComments/MethodInfoExtensions.cs

```diff
@@ -9,13 +9,65 @@ public static MethodInfo GetUnderlyingGenericTypeMethod(this MethodInfo construc
         {
             var constructedType = constructedTypeMethod.DeclaringType;
             var genericTypeDefinition = constructedType.GetGenericTypeDefinition();
+            var genericArguments = constructedType.GenericTypeArguments;
 
             // Retrieve list of candidate methods that match name and parameter count
             var candidateMethods = genericTypeDefinition.GetMethods()
                 .Where(m =>
                 {
-                    return (m.Name == constructedTypeMethod.Name)
-                        && (m.GetParameters().Length == constructedTypeMethod.GetParameters().Length);
+                    var genericTypeDefinitionParameters = m.GetParameters();
+                    var constructedTypeParameters = constructedTypeMethod.GetParameters();
```

Comment:
`constructedTypeParameters` can be hoisted outside of the LINQ method so it isn't computed more than once.